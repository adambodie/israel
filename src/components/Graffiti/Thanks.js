import React from 'react'
import { Column } from 'react-foundation'
import { connect } from 'react-redux'
import { pressButton } from '../../store/actions/thanks'

const mapStateToProps = state => {
	return {
		thanksButton: state.thanksButton,
	}
}
const mapDispatchToProps = { pressButton }

const Thanks = ({ thanksButton, pressButton }) => (
	<Column>
		<h1>What does this say?</h1>
		<div className='thanks'>
			<button data-testid='button-top' className='thanks-cover' style={{marginTop: thanksButton.top}} onClick={pressButton}>{thanksButton.text}</button>
			<img id='thanks-image' src='/assets/main/thanks.jpg' alt='thanks' />
		</div>
		<p>As you click on the button, you will see that that this isn't written in Arabic, as some would consider, but is actually a combination of Arabic on the top, and Hebrew on the bottom.  The word intended to be written is Thanks, which is written in Hebrew, Todah.  But the artist wasn't completely correct in the Arabic translation of Thanks, off by the last letter.  Shukrah is the translation of Thanks in Arabic.</p>
	</Column>
)

export default connect(mapStateToProps, mapDispatchToProps)(Thanks)