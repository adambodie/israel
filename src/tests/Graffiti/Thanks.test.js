import React from 'react'
import renderer from 'react-test-renderer'
import {
  render,
  cleanup,
  fireEvent
} from 'react-testing-library'

import Thanks from '../../components/Graffiti/Thanks'

afterEach(cleanup) 

it('renders correctly', () => {
	const tree = renderer.create(<Thanks />).toJSON()
	expect(tree).toMatchSnapshot()
})

it('calls "pressButton" prop on button click and changes text', () => {
    // Render new instance in every test to prevent leaking state
    const spy = jest.spyOn(Thanks.prototype, 'pressButton')
    const { getByTestId, rerender } = render(<Thanks />)
    fireEvent.click(getByTestId('button-top'))
    expect(spy).toHaveBeenCalled()
    rerender(<Thanks />)
    expect(getByTestId('button-top').textContent).toBe('Look down!')
  })