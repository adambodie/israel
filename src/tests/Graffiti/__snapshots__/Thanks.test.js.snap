// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="columns"
>
  <h1>
    What does this say?
  </h1>
  <div
    className="thanks"
  >
    <button
      className="thanks-cover"
      data-testid="button-top"
      onClick={[Function]}
      style={
        Object {
          "marginTop": 0,
        }
      }
    >
      Click Me!
    </button>
    <img
      alt="thanks"
      id="thanks-image"
      src="/assets/main/thanks.jpg"
    />
    <button
      className="thanks-cover"
      onClick={[Function]}
      style={
        Object {
          "marginTop": 310,
        }
      }
    >
      Click Me!
    </button>
  </div>
  <p>
    As you click on the blocks above and below the image, you will see that that this isn't written in Arabic, as some would consider, but is actually a combination of Arabic on the top, and Hebrew on the bottom.  The word intended to be written is Thanks, which is written in Hebrew, Todah.  But the artist wasn't completely correct in the Arabic translation of Thanks, off by the last letter.  Shukrah is the translation of Thanks in Arabic.
  </p>
</div>
`;
